### ISSUE TITLE
Investigate removing the globals __extends() and __assign() populated by applicationinsights-shims

### ISSUE THREAD

**Issue by MSNev** [microsoft/ApplicationInsights-JS#1280]
As part of removing tslib usage because of #1269, we created the applicationinsights-shims package, which to ensure that any modules that uses it also exposes the __extends() and __assign() onto the global scope (it reuses any existing instances).

We have had a request that at least for the ESM modules (further details will be provided) if we can not pollute the global scope with these functions.

**Comment by markwolff** [comment]
Seems like a good time to rip off the band aid and bump to typescript 3. WDYT?

**Comment by MSNev** [comment]
It's not that easy (as I already tried that before breaking the dependency), the issue is that we are targeting es3 and the "PropertyKey" added by tslib is not available when compiling for a es3 target.

I've not tried using v2 of tslib with TS 3.x yet though, but I expect the same issue (if not more) -- one of the reasons I called this "Investigate".

And we also need to make sure that the generated code / modules don't break older users with that sort of move :-(

**Comment by CraigComeOnThisNameCantBeTaken** [comment]
I am using Angular universal along with this package, and __extends and __assign are both undefined for me.

**Comment by kryalama** [comment]
@CraigComeOnThisNameCantBeTaken, can you please share the details of the "@microsoft/applicationinsights-web" version you are using in your code. We want to make sure "@microsoft/applicationinsights-shims": "^1.0.0" in your dependency list. Please refer  to suggestions from #1281.

**Comment by MSNev** [comment]
Did you use the applicationinsights-shims or the tslib version?

I have found (using a different modules) that Webpack ProviderPlugin can silently fail (and not include the references) if the module can't be found for some reason, which results in the same errors.

When the packages are correctly referenced and available the ProviderPlugin does work.

**Comment by MLoughry** [comment]
Perhaps I'm naïve, but what browsers are you trying to support by targeting `es3` instead of [`es5`](https://caniuse.com/#feat=es5)?

**Comment by MSNev** [comment]
Primarily IE8, we have customers that still need to target IE8 and a couple in the health industry that still need IE7.

**Comment by MLoughry** [comment]
Could you have a separate build under `lib-esm/` or similar that targets `es5` and `esnext` modules, then specify the path as [`"module":`](https://github.com/dherman/defense-of-dot-js/blob/master/proposal.md) in the `package.json`? This would be a backwards-compatible change that would allow apps targeting modern browsers to both optimize/tree-shake the package bundle themselves, and avoid the `tslib` workaround.

**Comment by MSNev** [comment]
The problem with that is many fold, however, this work-around actually has nothing to do with ES3 and our support for IE8.

The issue is one of compilation, we can't compile with the currently targeted version of TypeScript (2.5.3) AND use version 1.13.0 (or later) of tslib -- as fails to compile. The workaround was to break the dependency so that we don't break other teams / customers by targeting the older versions of tslib. 

In short, until we can successfully upgrade the version of typescript that we use (work in progress) without breaking other teams or customers we are limited to using this workaround... Unless of course if tslib publishes a fix.

**Comment by mmarshad** [comment]
@MSNev : installed the latest package but still having the issue
"@microsoft/applicationinsights-react-js": "^3.1.0",
 "@microsoft/applicationinsights-web": "^2.6.2",

**Comment by MSNev** [comment]
We have not yet published a react update that uses v2.0.0 of the shims (should be later today), if you check your dependencies you *should* find that v1.0.3 is currently still getting included.

**Comment by MSNev** [comment]
PR to update the versions for release is out #1552

**Comment by MSNev** [comment]
React v3.1.1 and ReactNative v2.3.1 are now published to NPM

**Comment by MSNev** [comment]
I've deprecated the react v3.1.1 and react-native v2.3.1 packages until I can publish an updated versions #1553

**Comment by MSNev** [comment]
v2.6.2 is now fully deployed

**Comment by MSNev** [comment]
I'm updating #1553 on the React and ReactNative releases with the current state, please see comments there for additional info.
